*** SDL_winmm.c.old	2016-11-12 00:47:40.000000000 +0300
--- SDL_winmm.c	2016-11-13 02:02:53.288203864 +0300
***************
*** 33,38 ****
--- 33,69 ----
  #include "../SDL_audio_c.h"
  #include "SDL_winmm.h"
  
+ /*====== WORKAROUND for MinGW's WinAPI header where those structures are being missed ======*/
+ typedef struct tagWAVEINCAPS2W 
+ {
+     WORD wMid;
+     WORD wPid;
+     MMVERSION vDriverVersion;
+     WCHAR szPname[MAXPNAMELEN];
+     DWORD dwFormats;
+     WORD wChannels;
+     WORD wReserved1;
+     GUID ManufacturerGuid;
+     GUID ProductGuid;
+     GUID NameGuid;
+ } WAVEINCAPS2W,*PWAVEINCAPS2W,*NPWAVEINCAPS2W,*LPWAVEINCAPS2W;
+ 
+ typedef struct tagWAVEOUTCAPS2W
+ {
+     WORD wMid;
+     WORD wPid;
+     MMVERSION vDriverVersion;
+     WCHAR szPname[MAXPNAMELEN];
+     DWORD dwFormats;
+     WORD wChannels;
+     WORD wReserved1;
+     DWORD dwSupport;
+     GUID ManufacturerGuid;
+     GUID ProductGuid;
+     GUID NameGuid;
+ } WAVEOUTCAPS2W,*PWAVEOUTCAPS2W,*NPWAVEOUTCAPS2W,*LPWAVEOUTCAPS2W;
+ /*==========================================================================================*/
+ 
  #ifndef WAVE_FORMAT_IEEE_FLOAT
  #define WAVE_FORMAT_IEEE_FLOAT 0x0003
  #endif
