*** SDL_audiotypecvt.c.old	2016-11-12 00:47:40.000000000 +0300
--- SDL_audiotypecvt.c	2016-11-13 00:09:35.244116864 +0300
***************
*** 239,245 ****
      SDL_memcpy(sample, src, cpy);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst >= target) {
          SDL_memcpy(dst, sample, cpy);
          dst -= 8;
          eps += srcsize;
--- 239,245 ----
      SDL_memcpy(sample, src, cpy);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst > target) {
          SDL_memcpy(dst, sample, cpy);
          dst -= 8;
          eps += srcsize;
***************
*** 320,326 ****
      SDL_assert(channels <= 8);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst >= target) {
          for (i = 0; i < channels; i++) {
              dst[i] = (float) ((((double)src[i]) + ((double)last_sample[i])) * 0.5);
          }
--- 320,326 ----
      SDL_assert(channels <= 8);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst > target) {
          for (i = 0; i < channels; i++) {
              dst[i] = (float) ((((double)src[i]) + ((double)last_sample[i])) * 0.5);
          }
***************
*** 355,361 ****
      SDL_assert(channels <= 8);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst >= target) {
          for (i = 0; i < channels; i++) {
              dst[i] = (float) ((((double) src[i]) + (3.0 * ((double) last_sample[i]))) * 0.25);
          }
--- 355,361 ----
      SDL_assert(channels <= 8);
      SDL_memcpy(last_sample, src, cpy);
  
!     while (dst > target) {
          for (i = 0; i < channels; i++) {
              dst[i] = (float) ((((double) src[i]) + (3.0 * ((double) last_sample[i]))) * 0.25);
          }
