From 3b07c8313a5988df3457e2be7c1c4c55e1922864 Mon Sep 17 00:00:00 2001
From: Vitaly Novichkov <admin@wohlnet.ru>
Date: Thu, 26 Sep 2019 23:50:27 +0300
Subject: [PATCH] Fixed build for Emscripten

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 73044c9..15d8e97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ if(NOT PNG_BUILD_ZLIB)
   include_directories(${ZLIB_INCLUDE_DIR})
 endif()
 
-if(UNIX AND NOT APPLE AND NOT BEOS AND NOT HAIKU)
+if(UNIX AND NOT APPLE AND NOT BEOS AND NOT HAIKU AND NOT EMSCRIPTEN)
   find_library(M_LIBRARY m)
 else()
   # libm is not needed and/or not available
-- 
2.17.1

